src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\Debug\ExceptionHandler;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * HttpExceptionListener.
+ *
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+          KernelEvents::EXCEPTION => array(
+            array('onKernelException', -256),
+          ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if ($event->hasResponse()) {
```

Comment:
Also I am not sure if this evaluates to true often in the production environment without twig installed.

Reply:
It should evaluate to false unless one handled the exception, either via twigbundle or any other mean, isn't it?

Reply:
We should forget about Twig here, it's totally independent code. What matters is if there is a listener that already set a response, and if not, then generate one. This question doesn't need to wonder about debug or not.
There are ppl that do set responses here, while not using twig (eg for APIs.)